app/assets/javascripts/application.js

```diff
@@ -21,6 +22,12 @@ document.addEventListener("turbolinks:load", function() {
       $(".js-select_all").prop('checked', false)
       $('button.archive_selected, button.unarchive_selected, button.mute_selected').addClass('hidden');
     }
+    var marked_unread_length = marked.parents('tr.active').length;
```

Comment:
I didn't want the button to show up when only read notifications were selected.  That makes for slightly more complexity, but better UX.